flow/layers/color_filter_layer.cc

```diff
@@ -38,6 +41,9 @@ void ColorFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
 }
 
 void ColorFilterLayer::Preroll(PrerollContext* context) {
+  auto mutator = context->state_stack.save();
+  mutator.translate(offset_);
+
```

Comment:
You need to adjust the paint bounds accordingly. The paint_bounds are measured in the parent coordinate system (which used to be a transform layer so it would be the agent applying the offset to the child's bounds to create its own bounds). But now that this layer is doing its own translation, it needs to adjust the bounds it reports.

Reply:
Done, but I noticed that there is a difference in how the image filter layer and this layer behave w.r.t child bounds transformation. I assume that is because matrix filters or blur can increase the size of things beyond the offseT?

Reply:
The bounds should always be offset because the translation is always applied in Paint. There is a bug in image_filter_layer in that it doesn't offset the bounds if there is no filter. Do not replicate that bug here the way you have - and fix IFL while you're at it if you please.

The bounds need to match what is rendered (modulo we haven't been replicating the integral transform code into the layer Preroll methods, which has been a topic of discussion with @knopp as we migrate Diff to use the LayerStateStack).

But, this offset is always applied in Paint and should always be applied to the paint bounds in both CFL and IFL.